### DISCUSSION TITLE
Config for local execution

### DISCUSSION THREAD

**Discussion by tbraeckevelt** [molmod/psiflow#35] [Q&A]
I am trying to run psiflow locally. I tried to not provide a config.yaml file, but that lead to ModuleNotFoundErrors, because it did not have the modules installed that were necessary to run certain apps.

So, I slightly modified the wq.yaml;

---
parsl_log_level: WARNING
default_threads: 4
container_engine: 'apptainer'
container_uri: 'oras://ghcr.io/molmod/psiflow:v4.0.0-rc1_rocm5.6'
ModelEvaluation:
  cores_per_worker: 4
  gpu: false
ModelTraining:
  cores_per_worker: 4
  gpu: true
  max_training_time: 1
  max_workers: 1
CP2K:
  cores_per_worker: 2
  max_evaluation_time: 0.3
  launch_command: 'apptainer exec -e --no-init oras://ghcr.io/molmod/cp2k:2023.2 /opt/entry.sh mpirun -np 2 -x OMP_NUM_THREADS=1 cp2k.psmp'
...

I get an assertion error if I set "gpu: false" in ModelTraining. Is it not possible to train (for debug purposes) on cpus?

**Comment by svandenhaute** [comment]
Initially, it seemed easier to just force `gpu: true` for training, but that does prevent you from being able to execute ridiculously short `model.train` calls when debugging.
In general though, `model.train` calls require at least a minute or so because they need to perform all the necessary imports, run through a nontrivial amount of torch code and be able to save a model to disk. If the timeout on training is set too short, some of these operations will not be able to proceed and you will end up introducing bugs in your code which would not be there in production settings where there is sufficient training time.

**Reply by svandenhaute** [reply]
So, to answer your question: it is not possible to train on cpus because that will only happen in debug scenarios, but in debug scenarios you really shouldn't be calling `model.train` because you risk creating more bugs purely because of the unphysically short execution times which then explore "edge" cases in the code...

**Reply by tbraeckevelt** [reply]
Thanks, that is clear.
